### PR TITLE
use phing for build process.

### PR DESCRIPTION
**Pull request by jeremygiberson**
This is a PHP library, lets make building easier for PHP devs by using the build tool for PHP.

Changes
- Removed `failonerror` as it is not compatible with Phing. 
- Added a `phpcbf` target for fixing the coding style issues.
- added `passthru = true` on execs to show output

### CODE REVIEW

**Comment by daylerees** [comment]
Hey Jeremy, thanks for the PR.

Since we're not using any phing-specific features, and the ant buildfile purely serves as a convenient wrapper for the test suite during development, I'm not entirely sure this is needed?

**Comment by jeremygiberson** [comment]
I think it is important for the purpose of fostering (and reducing difficulty of) contributing to this project. 

As an example, from my own experience:

I cloned this repo to make a contribution (still WIP). I wanted to make sure my code was conforming to the coding style guidelines of this project. I noticed phpcs was included. So, I did `vendor/bin/phpcs src` and there were many errors from files I didn't modify. 

After realizing that `build.xml` existed, I immediately tried `vendor/bin/phing` but nothing happened. I thought "Okay, maybe phing was installed globally on repo creators machine and they just forgot to require-dev it". No problem, I issue a global install of phing and try again, now I get an error message "failonerrors not supported by ExcTask". After some googling I determined this was likely an Ant build file. I'll skip the details on my failed attempt to install ant on my local (windows) machine--but that turned out to be a waste of time & dead end.

In the end, I settled for manually calling `phpcs` with the parameters configured in the build file. But that is too tedious to continue doing. 

So I opted to make build.xml compatible with the Phing build tool that will be installed w/out hassle for me when I do composer install.  Now everyone gets to enjoy the convenient wrapper for development. Not just people who have&use ant.

**Comment by danhanly** [comment]
Maybe the build process just needs to be documented a bit more clearly, thereby saving you from making incorrect assumptions. It doesn't necessarily mean the technology used is wrong, just that it wasn't immediately clear to you.

**Comment by jeremygiberson** [comment]
@danhanly I agree that the build process could be documented more clearly. But had it been documented clearly, I still propose that my PR makes the build process _better_. 

Because it swaps out `ant` for `phing` reducing the number of developer concerns to build.

With ant, to develop this package the developer has the following concerns
- git
- ant
- php
- composer

With phing, to develop this package the developer has the following concerns
- git
- php
- composer

That's one less developer concern, an improvement.

File: composer.json [hunk 1]
```diff
@@ -16,11 +16,17 @@
...
     },
     "autoload": {
         "psr-4": {
             "Scientist\\": "src"
         }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "": "tests"
+        }
```
Comment (daylerees) [inline composer.json]: What's this for?

Comment (HARRYKASIOR) [inline composer.json]: hi!
about possible fractals modelization 
observed in some works composed by WA Mozart

Comment (jeremygiberson) [inline composer.json]: Oops, I cherry-picked the build change out of a branch I'm working on and I must have missed this. I'll update the PR to remove this change.

**Comment by daylerees** [comment]
I don't use phing much, but I have no strong arguments. Thanks for the PR!

**Pull request merged**